sklearn/manifold/tests/test_t_sne.py

```diff
@@ -231,7 +235,7 @@ def test_binary_perplexity_stability():
             assert_array_almost_equal(P1, last_P1, decimal=4)
 
 
-def test_gradient():
+def test_gradient(global_dtype):
```

Comment:
It seems that `_kl_divergence` is converting the inputs to 32 bits.
Therefore, I don't think that we need to pass the `global_dtype` for this function.

We should probably preserver `dtype` and revisit these tests.

Reply:
For now, this test passes. I guess we can rework the test a bit once TSNE supports float64 (this is constrained by TSNE Cython routines which only operates on `float` at the moment).